lib/datastore.js

```diff
@@ -490,9 +497,12 @@ Datastore.prototype._update = function (query, updateQuery, options, cb) {
 	  return callback(err);
 	}
 
-	// Update the datafile
-    self.persistence.persistNewState(_.pluck(modifications, 'newDoc'), function (err) {
+    // Update the datafile
+	  var newDocs = _.pluck(modifications, 'newDoc');
+    self.persistence.persistNewState(newDocs, function (err) {
       if (err) { return callback(err); }
+      if (newDocs.length > 0)
+        self.emit('updated', newDocs);
```

Comment:
Although, this point does make me wonder if perhaps there should be an individual event fired for _each_ document rather than per document array, e.g.

```js
modifications.forEach(function(mod) {
  self.emit('updated', mod.newDoc, mod.oldDoc);
});
```

The same array enumerating would apply for the `'inserted'` and `'removed'` events as well.